jax/interpreters/xla.py

```diff
@@ -945,7 +966,11 @@ class DeviceArray:
   _HAS_DYNAMIC_ATTRIBUTES = True
 
   def __init__(self, aval: core.ShapedArray, device: Optional[Device],
-               lazy_expr: lazy.LazyExpr, device_buffer: PyLocalBuffer):
+               lazy_expr: lazy.LazyExpr,
+               device_buffer: PyLocalBuffer):
+    if isinstance(device_buffer, Sequence):
```

Comment:
The type signature of this function suggests `device_buffer` can only be a `PyLocalBuffer`, not a `Sequence`. Can we delete this `if ...` statement? If not, why the polymorphism?

Reply:
Done